### ISSUE TITLE
Can not write chore back to TM1 by Chore.from_json

### ISSUE THREAD

**Issue by Cubewise-Ronan** [cubewise-code/tm1py#948]
**Describe the bug**
We would like to update chores from one instance  to another one, but a popup error when _Chore.from_json_
<img width="782" alt="image" src="https://github.com/cubewise-code/tm1py/assets/65941927/6b8120aa-155a-4fda-86f9-d4383ae087b0">

**To Reproduce**
Export a chore
```
    with open(make_path(target_path, chore_name, "json"),
              "w",
              encoding="utf-8") as file:
        body = json.loads(chore.body)
        json.dump(body, file, indent=4)
```

Write it back, then it will popup an error
```
    with open(make_path(target_path, chore_name, "json"),
              "r",
              encoding="utf-8") as file:
        chore_json = json.loads(file.read())
        chore = Chore.from_json(json.dumps(chore_json))
        chore.name = chore_name
        tm1.chores.update_or_create(chore)
```

**Version**
TM1py 1.11

**Comment by rclapp** [comment]
You'll need to look at your json file. It is missing a key, "Step". You can use a python debugger to step through the code to see why.

Just looking at the ChoreTask class it expects every task to have an integer that specifies the order, called Step.

Chances are your export is not unpacking the the tasks completely



Sent from Workspace ONE Boxer



On Aug 2, 2023 8:11 PM, Ronan ***@***.***> wrote:

Describe the bug
We would like to update chores from one instance to another one, but a popup error when Chore.from_json
[image]<https://user-images.githubusercontent.com/65941927/257967092-6b8120aa-155a-4fda-86f9-d4383ae087b0.png>

To Reproduce
Export a chore

    with open(make_path(target_path, chore_name, "json"),
              "w",
              encoding="utf-8") as file:
        body = json.loads(chore.body)
        json.dump(body, file, indent=4)


Write it back, then it will popup an error

    with open(make_path(target_path, chore_name, "json"),
              "r",
              encoding="utf-8") as file:
        chore_json = json.loads(file.read())
        chore = Chore.from_json(json.dumps(chore_json))
        chore.name = chore_name
        tm1.chores.update_or_create(chore)


Version
TM1py 1.11

**Comment by Cubewise-Ronan** [comment]
Hi @rclapp ,

Thanks for your suggestion.

I fixed the issue by changing my code as below

```
def encode(target_path, chore_name, tm1):
    with open(make_path(target_path, chore_name, "json"),
              "r",
              encoding="utf-8") as file:
        chore_json = json.loads(file.read())
        for index, task in enumerate(chore_json.get("Tasks", [])):
            task["Step"] = index
            process_name = task.get("Process@odata.bind",
                                    "").replace("Processes('",
                                                "").replace("')", "")
            task["Process"] = {"Name": process_name}

        chore = Chore.from_json(json.dumps(chore_json))
        chore.name = chore_name
        tm1.chores.update_or_create(chore)
```

**Comment by MariusWirtz** [comment]
The problem is that TM1 returns a slightly different JSON when you read a chore than the JSON that must be posted when creating a chore.

I think we could make the `from_dict` function in `Chore` more robust. It could derive the correct value for the step itself and pass it into the `ChoreTask.from_dict` function.
I will look into this.

**Comment by MariusWirtz** [comment]
@Cubewise-Ronan 
I think I manged to resolve this issue. Please upgrade to the feature branch to test it.

```
pip uninstall tm1py
pip install https://github.com/cubewise-code/tm1py/archive/refs/heads/feature/make-chore-json-parser-more-robust.zip`

```

**Comment by MariusWirtz** [comment]
Closed with #950
